### DISCUSSION TITLE
Unable to build

### DISCUSSION THREAD

**Discussion by ltsoveranakin** [StarArawn/bevy_ecs_tilemap#514] [Q&A] [answered]
Hello, I am fairly new to rust, bevy and even more so to this plugin, so please do excuse any possible oversight of mine.

When I attempt to build with bevy_ecs_tilemap on version 0.12.0 and bevy on version 1.12.1, I get a few errors all of which state that "Crate-private type `RenderChunkSize` in public interface".

I do notice that bevy_ecs_tilemap 0.11.1 builds with no issue, however to my knowledge that is for the previous version of bevy.

For additional info: I'm using rust nightly on windows 11.

The error I get is as follows:
   --> C:\Users\rockstar\.cargo\registry\src\index.crates.io-6f17d22bba15001f\bevy_ecs_tilemap-0.12.0\src\render\chunk.rs:217:5
    |
217 |     pub render_size: RenderChunkSize,
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak crate-private type
    |
   ::: C:\Users\rockstar\.cargo\registry\src\index.crates.io-6f17d22bba15001f\bevy_ecs_tilemap-0.12.0\src\render\mod.rs:63:1
    |
63  | pub(crate) struct RenderChunkSize(UVec2);
    | --------------------------------- `RenderChunkSize` declared as crate-private

Any help would be greatly appreciated!

**Comment by rparrett** [comment, marked as answer]
It looks like https://github.com/StarArawn/bevy_ecs_tilemap/pull/512 may have introduced this issue.

As far as I can tell though, it only happens with rust versions older than 1.74. Could try upgrading to latest stable (1.76) or nightly?

Regardless, it seems like something that could probably be fixed on our end.

**Reply by ltsoveranakin** [reply]
This worked thanks! Turns out I was on an older version of rust nightly and running `rustup update` along with adding `nightly` to the `rust-toolchain` file solved my issue!
